core/domain/feedback_services_test.py

```diff
@@ -64,4 +65,169 @@ def test_status_of_newly_created_thread_is_open(self):
             EXP_ID, 'a_state_name', None, 'a subject', 'some text')
         threadlist = feedback_services.get_threadlist(EXP_ID)
         thread_status = threadlist[0]['status']
-        self.assertEqual(thread_status, feedback_models.STATUS_CHOICES_OPEN)   
+        self.assertEqual(thread_status, feedback_models.STATUS_CHOICES_OPEN)
+
+    def _generate_thread_id(self, exp_id):
+        return '1234'
+
+    def _create_suggestion_objects_for_tests(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 
+                       'generate_new_thread_id', self._generate_thread_id):
+            (feedback_services.
+                create_suggestion('exp_id1', 'author_id', 3,
+                                  'state_name', {'old_content': {}}))
+        with self.swap(feedback_models.FeedbackThreadModel, 
+                       'generate_new_thread_id', self._generate_thread_id):
+            (feedback_services.
+                create_suggestion('exp_id2', 'author_id', 3,
+                                  'state_name', {'old_content': {}}))
+
+    def test_create_and_get_suggestion(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 
+                       'generate_new_thread_id', self._generate_thread_id):
+            (feedback_services.
+                create_suggestion('exploration_id', 'author_id', 3,
+                                  'state_name', {'old_content': {}}))
+            suggestion = feedback_services.get_suggestion(
+                'exploration_id', '1234')
+            thread = feedback_models.FeedbackThreadModel.get(
+                'exploration_id.1234')
+            self.assertEqual(thread.status, 
+                             feedback_models.STATUS_CHOICES_OPEN)
+            self.assertEqual(suggestion.exploration_id, 'exploration_id')
+            self.assertEqual(suggestion.author_id, 'author_id')
+            self.assertEqual(suggestion.exploration_version, 3)
+            self.assertEqual(suggestion.state_name, 'state_name')
+            self.assertEqual(suggestion.state_content, {'old_content': {}})
+            self.assertEqual(suggestion.status, 
+                         feedback_models.SUGGESTION_STATUS_NEW)
+
+    def _get_pending_suggestion_threads(self, exploration_id):
+        thread1 = feedback_models.FeedbackThreadModel(
+            id='exp_id1.1234',
```

Comment:
Hm. I think this isn't quite the right approach in general (but sorry, I couldn't tell from earlier when we were chatting). The reason I'm concerned is because this requires lots of manual hard-coding and keeping things in sync. For example, if you change the thread id here you'll need to remember to change it in _get_suggestions() too.

I think, instead, a better approach might be to have a setUp() method, and in that method, create three feedback thread models with different properties.

Individual tests would then assume that the setUp has already been done, and could run get_*() methods on it, or change the status of individual threads, etc. Note that setUp() is run independently for each of the tests, so you don't have to worry about resetting the state of things.